doc/modules/grid_search.rst

```diff
@@ -227,317 +231,309 @@ need to explicitly import ``enable_halving_search_cv``::
+  iteration using all the resources. The exact value that is used depends on
+  the number of candidate parameter, on `max_resources` and on `factor`.
+
+  For :class:`HalvingRandomSearchCV`, exhausting the resources can be done in 2
+  ways:
+
+  - by setting `min_resources='exhaust'`, just like for
+    :class:`HalvingGridSearchCV`;
+  - by setting `n_candidates='exhaust'`.
+
+  Both options are mutually exclusive: using `min_resources='exhaust'` requires
+  knowing the number of candidates, and symmetrically `n_candidates='exhaust'`
+  requires knowing `min_resources`.
+
+  In general, exhausting the total number of resources leads to a better final
+  candidate parameter, and is slightly more time-intensive.
 
 .. _aggressive_elimination:
 
-Aggressive elimination of candidates
-------------------------------------
-
-Ideally, we want the last iteration to evaluate ``factor`` candidates (see
-:ref:`amount_of_resource_and_number_of_candidates`). We then just have to
-pick the best one. When the number of available resources is small with
-respect to the number of candidates, the last iteration may have to evaluate
-more than ``factor`` candidates::
-
-    >>> from sklearn.datasets import make_classification
-    >>> from sklearn.svm import SVC
-    >>> from sklearn.experimental import enable_halving_search_cv  # noqa
-    >>> from sklearn.model_selection import HalvingGridSearchCV
-    >>> import pandas as pd
-    >>>
-    >>>
-    >>> param_grid = {'kernel': ('linear', 'rbf'),
-    ...               'C': [1, 10, 100]}
-    >>> base_estimator = SVC(gamma='scale')
-    >>> X, y = make_classification(n_samples=1000)
-    >>> sh = HalvingGridSearchCV(base_estimator, param_grid, cv=5,
-    ...                          factor=2, max_resources=40,
-    ...                          aggressive_elimination=False).fit(X, y)
-    >>> sh.n_resources_
-    [20, 40]
-    >>> sh.n_candidates_
-    [6, 3]
-
-Since we cannot use more than ``max_resources=40`` resources, the process
-has to stop at the second iteration which evaluates more than ``factor=2``
-candidates.
-
-Using the ``aggressive_elimination`` parameter, you can force the search
-process to end up with less than ``factor`` candidates at the last
-iteration. To do this, the process will eliminate as many candidates as
-necessary using ``min_resources`` resources::
-
-    >>> sh = HalvingGridSearchCV(base_estimator, param_grid, cv=5,
-    ...                            factor=2,
-    ...                            max_resources=40,
-    ...                            aggressive_elimination=True,
-    ...                            ).fit(X, y)
-    >>> sh.n_resources_
-    [20, 20,  40]
-    >>> sh.n_candidates_
-    [6, 3, 2]
-
-Notice that we end with 2 candidates at the last iteration since we have
-eliminated enough candidates during the first iterations, using ``n_resources =
-min_resources = 20``.
+.. dropdown:: Aggressive elimination of candidates
```

Comment:
This section is being called [HalvingRandomSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.HalvingRandomSearchCV.html) and [HalvingGridSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.HalvingGridSearchCV.html) docstrings. We rather keep the section for traceability.

```suggestion
Aggressive elimination of candidates
------------------------------------

Using the ``aggressive_elimination`` parameter, you can force the search
process to end up with less than ``factor`` candidates at the last
iteration.

.. dropdown:: Code example of aggressive elimination
```